Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -14,6 +14,26 @@
  * limitations under the License.
  */
 
+/* NB: proto bytes were created via:
+     echo 'TEXT_FORMAT_PROTO' \
```

Comment:
I wonder if we could wrap this up in a little script we just check in. Constructing this command-line is tedious (though perhaps that's just because of all the plus signs in the diff).

Perhaps even that script should run at the higher level and just run through the cases we care about.

Feel free to push back.

Reply:
> I wonder if we could wrap this up in a little script we just check in. Constructing this command-line is tedious (though perhaps that's just because of all the plus signs in the diff).

In practice, constructing the command line is easy. (Just sweep your mouse over it... unless you're looking at a diff.) Also note the lack of '*' at the beginning of these lines to enable that.

Checking in a script would be reasonable. Alternatively, we could link the *test suite* (not the main sdk) against libprotobuf and use the normal protobuf api, or even the libprotobuf text format api to construct these protos.

> Perhaps even that script should run at the higher level and just run through the cases we care about.

i.e. driving the test from the script? So something like this:
```
#/bin/sh
set -e
for test in [ 'field:value' 'field2:value2' ] ; do
  echo 'field: value' | protoc --params | ./testEncodeDecode --maybeSomeParamsHere
done
```

I like that it wouldn't require us to record the raw bytes in the .cc file... but I'm less wild about driving the test from outside the normal mechanism. If we're going to build the bytes on the fly, I like linking libprotobuf better.

So I think our options are:
1) noop. Just keep building the bytes with protoc and adding them to the test suite.
2) drive this test with a script that looks something like the above (still hooked into cmake such that `make test` runs it)
3) link libprotobuf to test suite and use protobuf api OR text format proto api.

(Either way, if we decide to do 2 or 3, I'll issue a followup PR rather than including in this one.)

Reply:
sg